python/lsst/ts/tunablelaser/compoway_register.py

```diff
@@ -300,56 +306,59 @@ async def read_register_value(self):
+            )
 
-        with io.BytesIO(response) as f:
             try:
-                stx_node_subadd = f.read(5)
-                expected_stx_node_subadd = "\x02" + self.node + "\x30\x30"
+                stx_node_subadd = await self.component.commander.readexactly(5)
+                expected_stx_node_subadd = "\x02"
+                if int(self.node) < 10:
+                    expected_stx_node_subadd += "\x30"
+                expected_stx_node_subadd += self.node + "\x30\x30"
+                expected_stx_node_subadd = expected_stx_node_subadd.encode("UTF-8")
                 if stx_node_subadd is not expected_stx_node_subadd:
-                    raise Exception(
+                    self.log.error(
                         f"Received incorrect start of packet: {stx_node_subadd}, "
                         f"expected: {expected_stx_node_subadd}"
                     )
-                self.end_code = f.read(2)
-                mrc_src = f.read(4)
+                    self.register_value = ""
+                self.end_code = await self.component.commander.readexactly(2)
+                mrc_src = await self.component.commander.readexactly(4)
                 # read variable area request MRC is 01, SRC is 01
-                expected_mrc_src = "\x01\x01"
+                expected_mrc_src = b"\x30\x31\x30\x31"
                 if mrc_src is not expected_mrc_src:
-                    raise Exception(
+                    self.log.error(
                         f"Received incorrect Request Codes: {mrc_src}, "
                         f"expected: {expected_mrc_src}"
                     )
-                self.response_code = f.read(4)
-
-                # bytesio doesn't include a readuntil
-                # the cmdtxt can be variable length, demarked by ETX byte
-                # This read_elements setting only reads 1 word (4 bytes)
-                # If this needs to be configurable in the future one way
-                # is to make a dictionary like its done for register add
-                # Doing range 8 for comfort, should only be 4 + 1 for ETX
+                    self.register_value = ""
+                self.response_code = await self.component.commander.readexactly(4)
                 # TODO calculate maximum message length dynamically and
                 # TODO add for when f runs out of bytes
-                self.cmd_txt = b""
-                for _ in range(8):
-                    byte = f.read(1)
-                    # ETX byte
-                    if byte == b"\x03":
-                        break
-                    else:
-                        self.cmd_txt = self.cmd_txt + byte
+                self.cmd_txt = await self.component.commander.readuntil(b"\x03")
+                self.register_value = self.cmd_txt
 
-                self.BCC = f.read(1)
+                # trim off ETX byte
+                self.cmd_txt = self.cmd_txt[:-1]
 
+                self.BCC = await self.component.commander.readexactly(1)
```

Comment:
In python, we would usually use lower case for attributes, upper case is reserved for constants.